File: jimmer-ddd-java-spring-boot-starter/src/test/java/org/morecup/jimmerddd/java/spring/domain/goods/GoodsFactory.java
package org.morecup.jimmerddd.java.spring.domain.goods;

import org.morecup.jimmerddd.java.factory.FactoryEventHandler;
import org.morecup.jimmerddd.java.factory.WithFactoryContext;
import org.springframework.stereotype.Component;

@Component
public class GoodsFactory implements FactoryEventHandler {
    @WithFactoryContext
    public Goods createAndSave() {
        return Immutables.createGoods(draft -> {
            draft.setName("Goods");
            draft.setNowAddress("address");
        });
    }
}

File: jimmer-ddd-kotlin-spring-boot-starter/src/test/kotlin/org/morecup/jimmerddd/kotlin/spring/domain/goods/GoodsFactory.kt
package org.morecup.jimmerddd.kotlin.spring.domain.goods

import org.springframework.stereotype.Service

@Service
class GoodsFactory(
    private val goodsRepository: GoodsRepository
) {
    fun create(cmd: CreateGoodsCmd): Goods {
        return Goods {
            name = cmd.name
            nowAddress = cmd.nowAddress
        }
    }
}